src/utils/shared/expiration.js

```diff
@@ -0,0 +1,48 @@
+const moment = require('moment')
+
+const DEFAULT_EXPIRATION_OFFSET = { days: 7 }
+
+/*
+   Get the expiration date for an object with date and/or expires fields
+   If expires is given, return it as a moment
+   If expires is undefined, return date plus 7 days
+   If expires is false, return false, as the item never expires.
+*/
+function getExpirationDate({ date, expires }) {
+  if (expires === false) return null
+  if (expires) return moment(expires)
+  if (date) return moment(date).add(DEFAULT_EXPIRATION_OFFSET)
+  return null
+}
+
+/*
+   This is the primary logic to check if a date is expired,
+   It simply uses Moment to parse a date input and comparse that to the current
+   time.
+   Use this on the result of getExpirationDate to get both pieces of
+   information.
+*/
+function dateIsExpired(expires) {
+  return Boolean(expires && expires.isBefore(moment()))
```

Comment:
just curious - can && return undefined? do you need to wrap it with explicit Boolean( )?

Reply:
JS `&&` and `||` will always return the value it ended on, even on short-circuits.

For example, `null && true` will return `null`, not `false`. You can try it on the Node command line yourself, I just did to confirm.